### PR TITLE
Fix shaky code coverage

### LINKED ISSUE #551: Shaky code coverage in VirSatProjectCommons
There is still some shaky test coverage in VirSatProjectCommons. The method for collecting all VirSatProjects is running into sometimes open and non open projects. This may occur from some asynchronous updates triggering the VirSatNavigator.

image

The test cases should be improved

### PR DESCRIPTION
**Pull request by ghost**
- Now testing both variants with open and closed projects
- Additionally externalized method to check for correct project nature.
Method will be needed in server as well.

---
Closes #551: Shaky code coverage in VirSatProjectCommons

### CODE REVIEW

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/structure/VirSatProjectCommons.java [hunk 1]
```diff
@@ -318,6 +312,19 @@ public static void attachBuilder(IProject project, String builderId) throws Core
...
 
+	/**
+	 * Use this method to test if a workspace project has the Virtual Satellite Nature attached
+	 * @param project the project to be tested
+	 * @return true in case the Virtual Satellite project nature could be found.
+	 * @throws CoreException
+	 */
+	public static boolean isVirSatProject(IProject project) throws CoreException {
+		IProjectDescription description;
+		description = project.getDescription();
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/structure/VirSatProjectCommons.java:323]:
```suggestion
		IProjectDescription description = project.getDescription();

```

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
Looks good, just a minor cleanup request.

File: de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/structure/VirSatProjectCommonsTest.java [hunk 2]
```diff
@@ -189,6 +189,24 @@ public void testGetAllVirSatProjects() throws CoreException {
 		
 		assertThat("Project1 and 3 is contained", virSatProjects, hasItems(project1VirSat, project3VirSat));
 		assertThat("Project2 is not contained", virSatProjects, not(hasItem(project2Other)));
+		
+		// Close one project and check if it is not detected anymore
+		project1VirSat.close(null);
+		List<IProject> virSatProjects2 = VirSatProjectCommons.getAllVirSatProjects(ws);
+		assertThat("Project1 and 3 is contained", virSatProjects2, hasItems(project3VirSat));
+		assertThat("Project2 is not contained", virSatProjects2, not(hasItems(project2Other, project1VirSat)));
```
Comment (JAmmermann-DLR) [inline de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/structure/VirSatProjectCommonsTest.java:197]:
```suggestion
		assertThat("Project3 is contained", virSatProjects2, hasItems(project3VirSat));
		assertThat("Project1 and 2 is not contained", virSatProjects2, not(hasItems(project1VirSat, project2Other)));
```

**Review by ghost — CHANGES_REQUESTED** [review]
Some minor change

**Review by ghost — APPROVED** [review]

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Pull request merged**